core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```diff
@@ -77,13 +78,27 @@ public void build(AnalysisService analysisService) {
             tokenFilters.add(tokenFilter);
         }
 
-        int positionOffsetGap = analyzerSettings.getAsInt("position_offset_gap", 0);
-        int offsetGap = analyzerSettings.getAsInt("offset_gap", -1);
+        int positionIncrementGap = 0;
+        if (analyzerSettings.getAsMap().containsKey("position_offset_gap")){
+            if (Version.indexCreated(indexSettings).before(Version.V_2_0_0)){
+                if (analyzerSettings.getAsMap().containsKey("position_increment_gap")){
+                    throw new IllegalArgumentException("Option [position_offset_gap] in custom Analyzer [" + name() +
```

Comment:
Thanks @rjernst. I think this is what I'm doing here. 

1. Pre 2.0 index, we both accept ```position_offset_gap``` and ```position_increment_gap```. But if user specifies both, throw an exception.
2. Otherwise, if user still use ```position_offset_gap```, throw an exception. 

Do you think we should log some deprecation notice instead in the 2nd case? 
Or other improvement in the exception msg?